### PR TITLE
Fixed VariantOp crash in LLVMCPULowerExecutableTarget

### PR DESCRIPTION
**Pull request by abdulraheembeigh**

### CODE REVIEW

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
This seems OK, but you actually shouldnt need this cause this pass is defined as running on a ExecutableVariantOp . Can you give some idea of what error you are trying to fix.

Requesting changes, just to get more info cause you shouldnt need this.

**Comment by abdulraheembeigh** [comment]
> This seems OK, but you actually shouldnt need this cause this pass is defined as running on a ExecutableVariantOp . Can you give some idea of what error you are trying to fix.
> 
> Requesting changes, just to get more info cause you shouldnt need this.

I don't have an error but I wanted to ensure that the entry point checks about this.
Can we include an assert instead?

**Comment by benvanik** [comment]
It should never be possible for getOperation to return anything but a variant op, so the check or an assert is not need.

**Comment by benvanik** [comment]
(assertions are really useful for readability - the presence of the assertion indicates to a reader it's _possible_ for something to be false but they can also be confusing as if a reader saw what's effectively `assert(true && "should be true")` it'd then have a chance to create confusion: "wait, _can_ true be false? now I don't know" etc)

**Comment by benvanik** [comment]
(also also note that the pass manager verifies that passes declared as operating on a particular op or op interface are actually run there, see the pass definition saying it runs only on variant ops: https://github.com/openxla/iree/blob/e1614808e4ad2453429979ee119f9e1245b100e8/compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.td#L67-L69)

**Comment by abdulraheembeigh** [comment]
Thanks @benvanik , I've updated PR can please review it? :)

**Review by benvanik — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp [hunk 1]
```diff
@@ -158,7 +158,7 @@ static TilingConfig getTilingConfigForPipeline(ModuleOp moduleOp) {
 void LLVMCPULowerExecutableTargetPass::runOnOperation() {
   IREE::HAL::ExecutableVariantOp variantOp = getOperation();
   ModuleOp moduleOp = variantOp.getInnerModule();
-  if (!variantOp || !moduleOp) {
+  if (!moduleOp) {
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp:161]: @benvanik should this be made part of the variantOp verification instead?

Comment (benvanik) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp:161]: modules are optional as external variants (linking in external object files) have no modules - omitting the module should effectively disable codegen

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp:161]: Based on https://github.com/openxla/iree/pull/15277#discussion_r1370773874 maybe we just return silently here... For now an error is ok though.

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp (hunk 1 above)
Comment (benvanik) [reply compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp:161]: good point - well, this PR doesn't change the behavior, but we should look out for that as people start pushing in external dispatches (mostly GPU to start, as ukernels are best for CPU and those will have modules) - there must be something higher up opting out of these passes but it'd be good to have defense in depth

**Pull request merged**